### PR TITLE
Deploy doxygen docs to GitHub pages

### PR DESCRIPTION
**Pull request by dantti**

### CODE REVIEW

**Comment by dantti** [comment]
this is what most KDSomething are using https://kdab.github.io/KDSingleApplication/

**Comment by dantti** [comment]
I'll do a PR changing this then you can build locally and see if it pleases you

**Comment by dantti** [comment]
PR rebased result -> https://kdab.github.io/KDBindings/  once this is merged I'll remove my branch from the ones that are allowed to deploy so only what's in master will trigger a deploy

File: .github/workflows/documentation.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          sudo apt install -y doxygen
+
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Configure project
+        run: >
+          cmake -S . -B ./build -DKDBindings_DOCS=ON
```
Comment (LeonMatthesKDAB) [inline .github/workflows/documentation.yml:44]: Can you add a `docs` preset to the `CMakePresets.json`, so this becomes `cmake --preset=docs`
and the build call becomes `cmake --build --preset=docs` or `cmake --build --preset=docs --target docs`

**Comment by dantti** [comment]
hmm seems we are both pushing :)

**Review by LeonMatthesKDAB — COMMENTED** [review]
Small nitpick, then we can merge :)

File: .github/workflows/documentation.yml (hunk 1 above)
Comment (dantti) [reply .github/workflows/documentation.yml:44]: ok

**Review by LeonMatthesKDAB — APPROVED** [review]

**Pull request merged**